gate-web/src/main/groovy/com/netflix/spinnaker/gate/controllers/V2PipelineTemplatesController.java

```diff
@@ -0,0 +1,125 @@
+
+  @Autowired
+  public V2PipelineTemplatesController(V2PipelineTemplateService pipelineTemplateService,
+                                       TaskService taskService,
+                                       ObjectMapper objectMapper) {
+    this.pipelineTemplateService = pipelineTemplateService;
+    this.taskService = taskService;
+    this.objectMapper = objectMapper;
+  }
+
+  // TODO(jacobkiefer): Un-stub
+  @ApiOperation(value = "List pipeline templates.", response = HashMap.class, responseContainer = "List")
+  @RequestMapping(method = RequestMethod.GET)
+  public Collection<Map> list(@RequestParam(required = false) List<String> scopes) {
+    return null;
+  }
+
+  @ApiOperation(value = "Create a pipeline template.", response = HashMap.class)
+  @RequestMapping(method = RequestMethod.POST)
+  @ResponseStatus(value = HttpStatus.ACCEPTED)
+  public Map create(@RequestBody Map<String, Object> pipelineTemplate) {
+    return null;
+  }
+
+  @ApiOperation(value = "Resolve a pipeline template.", response = HashMap.class)
+  @RequestMapping(value = "/resolve", method = RequestMethod.GET)
+  public Map resolveTemplates(@RequestParam String source, @RequestParam(required = false) String executionId, @RequestParam(required = false) String pipelineConfigId) {
+    return null;
+  }
+
+  @ApiOperation(value = "Get a pipeline template.", response = HashMap.class)
+  @RequestMapping(value = "/{id}", method = RequestMethod.GET)
+  public Map get(@PathVariable String id) {
+    return null;
+  }
+
+  @ApiOperation(value = "Update a pipeline template.", response = HashMap.class)
+  @RequestMapping(value = "/{id}", method = RequestMethod.POST)
+  @ResponseStatus(value = HttpStatus.ACCEPTED)
+  public Map update(@PathVariable String id,
+                    @RequestBody Map<String, Object> pipelineTemplate,
+                    @RequestParam(value = "skipPlanDependents", defaultValue = "false") boolean skipPlanDependents) {
+    return null;
+  }
+
+  @ApiOperation(value = "Delete a pipeline template.", response = HashMap.class)
+  @RequestMapping(value = "/{id}", method = RequestMethod.DELETE)
+  @ResponseStatus(value = HttpStatus.ACCEPTED)
+  public Map delete(@PathVariable String id,
+                    @RequestParam(value = "application", required = false) String application) {
+    return null;
+  }
+
+  @ApiOperation(value = "List all pipelines that implement a pipeline template", response = List.class)
+  @RequestMapping(value = "/{id}/dependents", method = RequestMethod.GET)
+  public List<PipelineTemplateDependent> listPipelineTemplateDependents(
+    @PathVariable String id,
+    @RequestParam(value = "recursive", required = false) boolean recursive
+  ) {
+    return null;
+  }
+
+  private String getNameFromTemplate(PipelineTemplate template) {
+    return null;
+  }
+
+  private String getApplicationFromTemplate(PipelineTemplate template) {
+    return null;
+  }
+
+  private String encodeAsBase64(Object value) {
```

Comment:
Where are these coming from?

Reply:
Likely won't be used, removing.

Reply:
If the template is saved via an orca task (as they were for v1), orca will evaluate any spel expressions in the template.

Reply:
On a quick look, pipeline definitions are also base64 encoded as well likely for the same reason.

Reply:
Thanks for pointing that out Dan.